### PR TITLE
Tsdiffana quality checks

### PR DESCRIPTION
**Pull request by lesteve**
supersedes #97.

Creating the PR to get the ball rolling.

- [x] there is at least one TODO pertaining to the tsdiffana parameter documentation in a couple of functions. Suggestions welcome!
- [x] by removing plot_cv_tc completely I may have disregarded the session separating feature added in plot_cv_tc since
- [x] improve axis labels
- [x] there were some complaints about the colorbar during the Parietal meeting. I don't remember exactly what they were about unfortunately. Does anyone ?
- [x] tooltip for tsdiffana plots

### CODE REVIEW

**Comment by dohmatob** [comment]
- Looks like you guys some rebase issues (can't be auto-merged)
- Any progress on adding legends, etc. ? See issue #90.

**Comment by lesteve** [comment]
I rebased but I haven't checked that the tsdiffana reports work fine yet.

**Comment by dohmatob** [comment]
OK, nice.

A quick way to check would be to run: `ipython examples/nipype_preproc_spm_auditory.py --pdb`.
When it's done, point your browser (firefox) to the indicated output html file, then click on the preproc tab.

OK, I've just took a deep look at the code and output. There're are couple  of minor issues.
- failures when rotations present in affine header (solve using reorder_img)
- manual tweaking of dimensions of figures. We avoid this since the tweaking is done front-end in html
- use_same_figure param is not very useful
- It be nice to fill-in the `tsdiffana_tooltip` (it's marked TODO)

I'll roll a quick fix for all but the last issue and PR the changes to you.

**Comment by dohmatob** [comment]
@lesteve: refer to PR I sent you.

OK, this is the way things look now.

screenshot from 2015-05-06 07 38 45

But
- There is still the missing tooltip problem/
- The missing legends and labels

Hope this helps.

**Comment by dohmatob** [comment]
- Also, it'd be nice to have better titles than "slice_diff2_max_vol" or "diff2_mean_vol".

**Comment by dohmatob** [comment]
Once PR #108 is merged, you should be fine.

**Comment by dohmatob** [comment]
Any update ?

**Comment by lesteve** [comment]
I rebased on master but I still have a check_niimg that we added and needs fixing. I probably won't have time to look at this before next week.

File: pypreprocess/time_diff.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+        ax.axis((0, val) + xlims[2:])
+        ax.set_xlabel(xlabel)
+        ax.set_ylabel(ylabel)
+
+    iter_axes = iter(axes)
+
+    # plot of mean volume variance
+    ax = next(iter_axes)
+    ax.plot(results['volume_mean_diff2'] / mean_means)
+    xmax_labels(ax, T - 1, 'Difference image number', 'Scaled variance')
```
Comment (lesteve) [inline pypreprocess/time_diff.py]: @bthirion all the xmax_labels calls are where you can improve of the axes titles.

**Comment by dohmatob** [comment]
OK, looks good. Only thing left now is the tootips explaining the thumbnails the user is looking at. We don't need these to be over-elaborate. Once this is done, it's a merge.

**Comment by dohmatob** [comment]
* Also, it'd be nice to have broken vertical lines seperating 1 vertical line from the other.  Take a look at https://github.com/neurospin/pypreprocess/blob/master/pypreprocess/reporting/check_preprocessing.py lines 141 through 144, for example.

**Comment by lesteve** [comment]
> Only thing left now is the tooltips explaining the thumbnails the user is looking at

I updated the todo list at the top of this issue. I think there are a few more things remaining though.

**Comment by bthirion** [comment]
I've sent you  a PR for some minor aspects.

**Comment by lesteve** [comment]
I just added the session separation, any comments let me know!

pypreprocess_session_separation

Looks like I will need to rebase again but other than that I reckon all the todo list items have been tackled.

**Comment by dohmatob** [comment]
Nice! We still have the tooltip issue though (we need a different /
specific tooltip string per thumbnail).

On Mon, Jun 1, 2015 at 10:43 AM, Loïc Estève <notifications@github.com>
wrote:

> I just added the session separation, any comments let me know!
>
> [image: pypreprocess_session_separation]
> <https://cloud.githubusercontent.com/assets/1680079/7909207/e04b1e72-084a-11e5-882c-29a68f06e688.png>
>
> Looks like I will need to rebase again but other than that I reckon all
> the todo list items have been tackled.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/neurospin/pypreprocess/pull/100#issuecomment-107368848>
> .
>



-- 
DED

**Comment by lesteve** [comment]
> Nice! We still have the tooltip issue though (we need a different / specific tooltip string per thumbnail).

But that can be done separately from this PR I reckon.

**Comment by dohmatob** [comment]
No, it can't be done in a separate PR. The PR is not complete without this. Otherwise the thumbs would be meaningless.

**Comment by bthirion** [comment]
On 01/06/2015 11:27, DOHMATOB Elvis wrote:
>
> No, it can't be done in a separate PR. The PR is not complete without 
> this. Otherwise the thumbs would be meaningless.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/neurospin/pypreprocess/pull/100#issuecomment-107379672>.
>
But I've sent a PR to Loic, haven't I ?

B

**Comment by dohmatob** [comment]
OK, cool. So I guess we'll be fine once @bthirion's PR (to @lesteve)  is merged into this PR.

File: pypreprocess/subject_data.py [hunk 2]
```diff
@@ -39,14 +40,23 @@
...
-                 "begining of the acquisition, due to saturation effects, "
-                 "scanner instability, etc.")
+tsdiffana_tooltips = [
+    ("(Squared) differences across sequential volumes. "
+     "A large value indicates an artifact "
+     "that occurred during the slice acquisition, possibly related to motion."),
+    ("Average signal over each volume. A large drop/peak (e.g. 5%) indicates "
+     "an artefact."),
+    ("Variance index per slice. Note that aqquisition artifacts can be slice"
+     "-specific. Look at the day if there is a peak somewhere."),
```
Comment (lesteve) [inline pypreprocess/subject_data.py]: @bthirion "Look at the day" :confused:. Should it be "Look at the data instead"?

**Comment by dohmatob** [comment]
OK, things look much better now. I'll merge and then we'll improve things as we go along.

File: pypreprocess/subject_data.py (hunk 2 above)
Comment (dohmatob) [inline pypreprocess/subject_data.py]: Fixed ;)

**Pull request merged**